lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -2254,19 +2254,19 @@ class SqlParser(private val ion: IonSystem) : Parser {
     }
 
     private fun ParseNode.throwTopLevelParserError(): Nothing =
-        token?.err("Keyword $token only expected at the top level in the query", PARSE_UNEXPECTED_TERM)
-        ?: throw ParserException("Keyword $token only expected at the top level in the query", PARSE_UNEXPECTED_TERM, PropertyValueMap())
+        token?.err("Keyword ${token.text} only expected at the top level in the query", PARSE_UNEXPECTED_TERM)
+        ?: throw ParserException("Keyword ${token?.text} only expected at the top level in the query", PARSE_UNEXPECTED_TERM, PropertyValueMap())
 
     /**
-     * Validates tree to make sure that the top level tokens are not found below the top level
+     * Validates tree to make sure that the top level tokens are not found below the top level.
+     * Top level tokens are the tokens or keywords which are valid to be used only at the top level in the query.
+     * i.e. these tokens cannot be used with a mix of other commands. Hence if more than one top level tokens are found
+     * in the query then it is invalid.
+     * [level] is the current traversal level in the parse tree.
+     * If [topLevelTokenSeen] is true, it means it has been encountered at least once before while traversing the parse tree.
```

Comment:
nit: description of `topLevelTokenSeen` could be more more specific. Consider "...it means a top level token has been previously encountered while traversing the parse tree"

Reply:
I believe the variable name itself tells it all.